### ISSUE TITLE
As a CIF administrator, I don't want payment amounts this milestones to be calculated before report received dates and eligible expenses have been filled out, so that I don't see erroneous data on the form

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-cif#1514]
#### Description:

Split from #1480

**For Advanced Milestone:**
- Calculate and display the "Gross/Net/Holdback Payment Amounts This Milestone" only after the "Report Received Date" field is filled out. Otherwise, display "This field cannot be calculated due to lack of information now.".

**For General Milestone:**
- Calculate and display the "Gross/Net/Holdback Payment Amounts This Milestone" only after the "Total Eligible Expenses" and "Report Received Date" fields are filled out. Otherwise, display "This field cannot be calculated due to lack of information now.".

#### Acceptance Criteria:

Given I am a CIF administrator
When I edit an advance milestone
And "Report Received Date" is empty
Then the "Gross/Net/Holdback Payment Amounts This Milestone" fields display "This field cannot be calculated due to lack of information now."

Given I am a CIF administrator
When I edit an advance milestone
And "Report Received Date" is filled out
Then the "Gross/Net/Holdback Payment Amounts This Milestone" fields are calculated.

Given I am a CIF administrator
When I edit a general milestone
And either of "Report Received Date" or "Total Eligible Expenses" is empty
Then the "Gross/Net/Holdback Payment Amounts This Milestone" fields display "This field cannot be calculated due to lack of information now."

Given I am a CIF administrator
When I edit a general milestone
And both "Report Received Date" and "Total Eligible Expenses" are filled out
Then the "Gross/Net/Holdback Payment Amounts This Milestone" fields are calculated.

#### Development Checklist:

- [x] update computed columns for the calculated values, to return null in the above cases
- [x] update tests
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1514?planning-poker) with Zenhub @BCerki @dleard @pbastia @Sepehr-Sobhani @tmastrom @sam-warren

**Comment by LindsayMacfarlane** [comment]
@dleard - please add your estimate

**Comment by pbastia** [comment]
Confirming this is in -test and working as expected! Thank you @Sepehr-Sobhani

**Comment by pbastia** [comment]
Thanks @Sepehr-Sobhani , excellent work!
A couple of issues:
- changing the milestone type doesn't update the calculations
- existing data on load (see dev data) doesn't initialize the form with the right values properly

**Comment by pbastia** [comment]
Marking this as complete, @Sepehr-Sobhani to write a bug card to address those issues.

**Comment by Sepehr-Sobhani** [comment]
Bug card created: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1628
